pkg/driverbuilder/builder/builders.go

```diff
@@ -70,33 +74,67 @@ func (c Config) ToProbeFullPath() string {
 }
 
 type commonTemplateData struct {
-	DriverBuildDir    string
-	ModuleDownloadURL string
-	ModuleDriverName  string
-	ModuleFullPath    string
-	BuildModule       bool
-	BuildProbe        bool
-	GCCVersion        string
-	CmakeCmd          string
+	DriverBuildDir   string
+	ModuleDriverName string
+	ModuleFullPath   string
+	BuildModule      bool
+	BuildProbe       bool
+	GCCVersion       string
+	CmakeCmd         string
 }
 
 // Builder represents a builder capable of generating a script for a driverkit target.
 type Builder interface {
 	Name() string
+	TemplateKernelUrlsScript() string
 	TemplateScript() string
 	URLs(kr kernelrelease.KernelRelease) ([]string, error)
-	TemplateData(c Config, kr kernelrelease.KernelRelease, urls []string) interface{} // error return type is managed
+	KernelTemplateData(kr kernelrelease.KernelRelease, urls []string) interface{} // error return type is managed
 }
 
-// MinimumURLsBuilder is an optional interface
+// MinimumURLsBuilder is an optional interface implemented by builders
 // to specify minimum number of requested headers urls
 type MinimumURLsBuilder interface {
 	MinimumURLs() int
 }
 
-func Script(b Builder, c Config, kr kernelrelease.KernelRelease) (string, error) {
-	t := template.New(b.Name())
-	parsed, err := t.Parse(b.TemplateScript())
+// TemplateDataSpecifier is an optional interface implemented by builders
+// to specify a custom template data instead of the default one.
+type TemplateDataSpecifier interface {
```

Comment:
`TemplateData` method is now optional since `config.ToTemplateData` is enough most of the time (for now, the only exception is `local` builder; and i think it will remain it).